src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1197 @@
+
+        text_len = text_len if text_len is not None else input_ids.size(1)
+        image_len = self.beit3.vision_embedding.num_position_embeddings()
+        max_len = text_len + image_len
+        uni_mask = torch.zeros((max_len, max_len), dtype=torch.long, device=input_ids.device)
+        i_start, i_end = 0, image_len
+        t_start, t_end = image_len, max_len
+        # triangle mask for caption to caption
+        uni_mask[t_start:t_end, t_start:t_end] = torch.tril(
+            torch.ones(text_len, text_len, dtype=torch.long, device=input_ids.device)
+        )
+        # full attention for caption to image
+        uni_mask[t_start:t_end, i_start:i_end] = 1
+        # full attention for image to image
+        uni_mask[i_start:i_end, i_start:i_end] = 1
+        uni_mask = 1 - uni_mask
+
+        if incremental_state is not None:
+            for idx in range(self.get_num_layers()):
+                if idx not in incremental_state:
+                    incremental_state[idx] = {}
+
+        # for incremental decoding
+        positions = None
+        if pixel_values is None:
+            uni_mask = uni_mask[-2:]
+            padding_mask = None
+            # start position (2 (fairseq starts at 2) + cur_position) is equal to text_len
+            positions = (
+                torch.arange(text_len, input_ids.size(1) + text_len, device=input_ids.device).long().unsqueeze(0)
+            )
+
+        outputs = self.beit3(
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            text_padding_position=padding_mask,
+            attn_mask=uni_mask,
+            incremental_state=incremental_state,
+            positions=positions,
+        )
+        if pixel_values is not None:
+            text_feats = outputs.encoder_out[:, image_len:]
+        else:
+            text_feats = outputs.encoder_out
+
+        if language_masked_pos is not None:
+            text_feats = text_feats[language_masked_pos.bool()]
+
+        logits = self.mlm_classifier(text_feats)
+
+        loss = None
+        if labels is not None:
+            # we are doing next-token prediction; shift prediction scores and input ids by one
+            eps = self.label_smoothing
+            n_class = logits.size(1)
+            one_hot = torch.zeros_like(logits).scatter(1, labels.view(-1, 1), 1)
+            one_hot = one_hot * (1 - eps) + (1 - one_hot) * eps / (n_class - 1)
+            log_prb = self.log_soft(logits)
+            loss = self.kl(log_prb, one_hot).sum(1)
+
+        if not return_dict:
+            output = (logits,)
+            output = output + (outputs.hidden_states,) if output_hidden_states else output
+            return ((loss.mean(),) + output) if loss is not None else output
+
+        return CausalLMOutputWithPast(
+            loss=loss.mean() if loss is not None else None,
+            logits=logits,
+            hidden_states=outputs.hidden_states,
+        )
+
+
+class Pooler(nn.Module):
```

Comment:
All custom layers should have the model prefix

```suggestion
class Beit3Pooler(nn.Module):
```